File: bench/sieve/sieve.js
// Sieve of Eratosthenes: tight loops over a large flat array.
const limit = 4000000;

const flags = new Array(limit + 1).fill(true);

let count = 0;
for (let p = 2; p <= limit; p++) {
  if (flags[p]) {
    count++;
    for (let m = p * p; m <= limit; m += p) {
      flags[m] = false;
    }
  }
}

console.log(count);

File: bench/sieve/sieve.py
# Sieve of Eratosthenes: tight loops over a large flat array.
limit = 4000000

flags = [True] * (limit + 1)

count = 0
for p in range(2, limit + 1):
    if flags[p]:
        count += 1
        m = p * p
        while m <= limit:
            flags[m] = False
            m += p

print(count)
